File: src/alfianLib/fn.java
package alfianLib;

import java.math.BigDecimal;
import java.math.BigInteger;
import java.util.Scanner;

public class fn extends mo {

    private static Scanner scan;
    /* Input */

    public static String inStr() { return scan.next(); }

    public static String inStr(String word) { out(word); return scan.next(); }

    public static String inLn() { return scan.nextLine(); }

    public static String inLn(String word) { out(word); return scan.nextLine(); }

    public static int inInt() { return scan.nextInt(); }

    public static int inInt(String word) { out(word); return scan.nextInt(); }

    public static BigInteger inBI() { return scan.nextBigInteger(); }

    public static BigInteger inBI(String word) { out(word); return scan.nextBigInteger(); }

    public static double inDbl() { return scan.nextDouble(); }

    public static double inDbl(String word) { out(word); return scan.nextDouble(); }

    public static long inLg() { return scan.nextLong(); }

    public static long inLg(String word) { out(word); return scan.nextLong(); }

    public static long inByte() { return scan.nextByte(); }

    public static long inByte(String word) { out(word); return scan.nextByte(); }

    public static BigDecimal inBD() { return scan.nextBigDecimal(); }

    public static BigDecimal inBD(String word) { out(word); return scan.nextBigDecimal(); }

    /* Output */
    public static void out() { System.out.print(""); }

    public static void out(char c) { System.out.print(c); }

    public static void out(String s) { System.out.print(s); }

    public static void out(int i) { System.out.print(i); }

    public static void out(Double d) { System.out.print(d); }

    public static void out(String[] arrStr) {
        for (String s : arrStr) {
            System.out.print(s + " ");
        } System.out.println();
    }

    public static void out(int[] arrInt) {
        for (int i : arrInt) {
            System.out.print(i + " ");
        } System.out.println();
    }

    public static void out(String[] arrStr, int[] arrInt) {
        int totalLength = arrStr.length + arrStr.length; for (int i = 0; i < totalLength; i++) {
            if (i < arrStr.length && i < arrInt.length) {
                System.out.print(arrStr[i] + " " + arrInt[i] + " ");
            }
        } System.out.println();
    }

    public static void out(int[] arrInt, String[] arrStr) {
        int totalLength = arrStr.length + arrInt.length; for (int i = 0; i < totalLength; i++) {
            if (i < arrStr.length && i < arrInt.length) {
                System.out.print(arrInt[i] + " " + arrStr[i] + " ");
            }
        } System.out.println();
    }

    public static void outln() { System.out.println(); }

    public static void outln(String s) { System.out.println(s); }

    public static void outln(int i) { System.out.println(i); }

    public static void outln(Double d) { System.out.println(d); }

    public static void outln(String[] arrStr) {
        for (String s : arrStr) {
            System.out.println(s);
        }
    }

    public static void outln(int[] arrInt) {
        for (int i : arrInt) {
            System.out.println(i);
        }
    }

    public static void outln(int[] arrInt, String[] arrStr) {
        int totalLength = arrStr.length + arrInt.length; for (int i = 0; i < totalLength; i++) {
            if (i < arrStr.length && i < arrInt.length) {
                System.out.println(arrInt[i] + " " + arrStr[i] + " ");
            }
        }
    }

    public static void outln(String[] arrStr, int[] arrInt) {
        int totalLength = arrStr.length + arrInt.length; for (int i = 0; i < totalLength; i++) {
            if (i < arrStr.length && i < arrInt.length) {
                System.out.println(arrStr[i] + " " + arrInt[i] + " ");
            }
        }
    }
}

File: src/alfianLib/Fn.kt
package alfianLib

import java.math.BigDecimal
import java.math.BigInteger
import java.util.*

object Fn : mo() {
    private val scan: Scanner? = null
    /* Input */
    fun inStr(): String {
        return scan!!.next()
    }

    fun inStr(word: String?): String {
        out(word)
        return scan!!.next()
    }

    fun inLn(): String {
        return scan!!.nextLine()
    }

    fun inLn(word: String?): String {
        out(word)
        return scan!!.nextLine()
    }

    fun inInt(): Int {
        return scan!!.nextInt()
    }

    fun inInt(word: String?): Int {
        out(word)
        return scan!!.nextInt()
    }

    fun inBI(): BigInteger {
        return scan!!.nextBigInteger()
    }

    fun inBI(word: String?): BigInteger {
        out(word)
        return scan!!.nextBigInteger()
    }

    fun inDbl(): Double {
        return scan!!.nextDouble()
    }

    fun inDbl(word: String?): Double {
        out(word)
        return scan!!.nextDouble()
    }

    fun inLg(): Long {
        return scan!!.nextLong()
    }

    fun inLg(word: String?): Long {
        out(word)
        return scan!!.nextLong()
    }

    fun inByte(): Long {
        return scan!!.nextByte().toLong()
    }

    fun inByte(word: String?): Long {
        out(word)
        return scan!!.nextByte().toLong()
    }

    fun inBD(): BigDecimal {
        return scan!!.nextBigDecimal()
    }

    fun inBD(word: String?): BigDecimal {
        out(word)
        return scan!!.nextBigDecimal()
    }

    /* Output */
    fun out() {
        print("")
    }

    fun out(c: Char) {
        print(c)
    }

    fun out(s: String?) {
        print(s)
    }

    fun out(i: Int) {
        print(i)
    }

    fun out(d: Double?) {
        print(d)
    }

    fun out(arrStr: Array<String>) {
        for (s in arrStr) {
            print("$s ")
        }
        println()
    }

    fun out(arrInt: IntArray) {
        for (i in arrInt) {
            print("$i ")
        }
        println()
    }

    fun out(arrStr: Array<String>, arrInt: IntArray) {
        val totalLength = arrStr.size + arrStr.size
        for (i in 0 until totalLength) {
            if (i < arrStr.size && i < arrInt.size) {
                print(arrStr[i] + " " + arrInt[i] + " ")
            }
        }
        println()
    }

    fun out(arrInt: IntArray, arrStr: Array<String>) {
        val totalLength = arrStr.size + arrInt.size
        for (i in 0 until totalLength) {
            if (i < arrStr.size && i < arrInt.size) {
                print(arrInt[i].toString() + " " + arrStr[i] + " ")
            }
        }
        println()
    }

    fun outln() {
        println()
    }

    fun outln(s: String?) {
        println(s)
    }

    fun outln(i: Int) {
        println(i)
    }

    fun outln(d: Double?) {
        println(d)
    }

    fun outln(arrStr: Array<String?>) {
        for (s in arrStr) {
            println(s)
        }
    }

    fun outln(arrInt: IntArray) {
        for (i in arrInt) {
            println(i)
        }
    }

    fun outln(arrInt: IntArray, arrStr: Array<String>) {
        val totalLength = arrStr.size + arrInt.size
        for (i in 0 until totalLength) {
            if (i < arrStr.size && i < arrInt.size) {
                println(arrInt[i].toString() + " " + arrStr[i] + " ")
            }
        }
    }

    fun outln(arrStr: Array<String>, arrInt: IntArray) {
        val totalLength = arrStr.size + arrInt.size
        for (i in 0 until totalLength) {
            if (i < arrStr.size && i < arrInt.size) {
                println(arrStr[i] + " " + arrInt[i] + " ")
            }
        }
    }
}